code/modules/reagents/Chemistry-Recipes.dm

```diff
@@ -79,7 +79,11 @@
 		if(L.stat != DEAD)
 			e.amount *= 0.5
 	e.start()
-	holder.clear_reagents()
+	if(ismob(holder))
```

Comment:
Why does waterpot still work for non-mobs?

Reply:
From a logical standpoint it doesn't make sense, from a balance standpoint chems in a body are rather important, so shouldn't be so easily deleted.

Reply:
Yeah, but why *only* fix it for mobs?

Reply:
In a container it can be assumed it's open, so the contents would be sprayed everywhere, otherwise the reagents would just get jostled around a bit.

That's a fair point. Why not have it check for opencontainer instead of ismob(holder)?

Reply:
Even better. Rather than delete the reagents should it be an opencontainer, make reagent puddles in the surrounding area, let's say about the light range of the explosion

Reply:
I'm thinking about stuff like clearing reagents from food/the TEG/other future uses/it's inconvenient when it happens accidentally

Reply:
Vomit puddles contain vomit in addition to a portion of the chems inside somebody.

Reply:
Oh, that's pretty rad.

Reply:
If it's about blowing the reagents out of an open container, it should check `OPENCONTAINER`, not `ismob`
That still seems like a pretty arbitrary distinction, but better than this

Reply:
done